AlembicImporter/Assets/UnitTests/AlembicExportTest.cs

```diff
@@ -126,110 +116,369 @@ public virtual void Term()
+
+        private AlembicExporter GetAlembicExporter () {
+            var alembicExporter = Object.FindObjectOfType<AlembicExporter> ();
+            Assert.That (alembicExporter, Is.Not.Null);
             return alembicExporter;
         }
 
-        private string GetAssetsAbsolutePath(string relPath)
-        {
-            return Application.dataPath + relPath.Replace("Assets", "");
+        private string GetAssetsAbsolutePath (string relPath) {
+            return Application.dataPath + relPath.Replace ("Assets", "");
         }
 
         /// <summary>
         /// Test that the abc file was reimported and can be loaded into Unity through
         /// the asset database.
         /// </summary>
         /// <param name="abcPath"></param>
-        private void TestAbcImported(string abcPath)
-        {
-            AssetDatabase.Refresh();
+        private void TestAbcImported (string abcPath) {
+            AssetDatabase.Refresh ();
 
-            var absPath = GetAssetsAbsolutePath(abcPath);
+            var absPath = GetAssetsAbsolutePath (abcPath);
 
-            Assert.That(string.IsNullOrEmpty(absPath), Is.False);
-            Assert.That(File.Exists(absPath));
+            Assert.That (string.IsNullOrEmpty (absPath), Is.False);
+            Assert.That (File.Exists (absPath));
 
             // now try loading the asset to see if it imported properly
-            var obj = AssetDatabase.LoadMainAssetAtPath(abcPath);
-            Assert.That(obj, Is.Not.Null);
+            var obj = AssetDatabase.LoadMainAssetAtPath (abcPath);
+            Assert.That (obj, Is.Not.Null);
             var go = obj as GameObject;
-            Assert.That(go, Is.Not.Null);
+            Assert.That (go, Is.Not.Null);
+        }
+
+        // Loads a given scene, generates a random export path and sends out the alembic exporter
+        void GenericSceneLoader (string sceneToLoad) {
+            SceneManagement.EditorSceneManager.LoadScene (sceneToLoad, UnityEngine.SceneManagement.LoadSceneMode.Single);
+        }
+
+        // Selects the Exporter in the scene and creates a path for it
+        void GenericExporterSelector () {
+            selectedExporter = GetAlembicExporter ();
+            var exportPath = GetRandomAbcFilePath ();
+            selectedExportPath = exportPath;
+            selectedExporter.recorder.settings.OutputPath = GetAssetsAbsolutePath (selectedExportPath);
+        }
+
+        // One shot export
+        [UnityTest]
+        public IEnumerator TestOneShotExport () {
+
+            GenericSceneLoader (c_scene);
+            yield return null;
+            GenericExporterSelector ();
+            yield return null;
+            selectedExporter.OneShot ();
+            yield return null;
+            TestAbcImported (selectedExportPath);
+        }
+
+        // Export Cloth
+        [UnityTest]
+        public IEnumerator TestClothExport () {
+            GenericSceneLoader (c_scene);
```

Comment:
this is a good start but I think we can clean it up further. I've written some helper functions:
```
        IEnumerator LoadScene(string sceneToLoad)
        {
            SceneManagement.EditorSceneManager.LoadScene(sceneToLoad, UnityEngine.SceneManagement.LoadSceneMode.Single);
            yield return null;
        }

        string SetupExporter(AlembicExporter exporter = null)
        {
            if (exporter == null) {
                exporter = GetAlembicExporter();
            }
            var exportPath = GetRandomAbcFilePath();
            exporter.recorder.settings.OutputPath = GetAssetsAbsolutePath(exportPath);
            return exportPath;
        }

        IEnumerator RecordAlembic(AlembicExporter exporter = null)
        {
            if (exporter == null)
            {
                exporter = GetAlembicExporter();
            }
            exporter.BeginRecording();

            while (!exporter.recorder.recording)
            {
                yield return null;
            }

            while (exporter.recorder.recording)
            {
                yield return null;
            }

        }
```

So TestClothExport would become:
```
        [UnityTest]
        public IEnumerator TestClothExport () {

            yield return LoadScene(c_scene);
            var exporter = GetAlembicExporter();
            var exportFile = SetupExporter(exporter);

            // set test specific exporter settings here
            exporter.maxCaptureFrame = 150;

            yield return RecordAlembic(exporter);
            TestAbcImported(exportFile);
    }
```
Then since a lot of the tests are the same, but just using a different scene maybe we can go further and do something like this:
```
        IEnumerator TestScene(string scene)
        {
            yield return LoadScene(scene);
            var exporter = GetAlembicExporter();
            var exportFile = SetupExporter(exporter);

            // set test specific exporter settings here
            exporter.maxCaptureFrame = 150;

            yield return RecordAlembic(exporter);
            TestAbcImported(exportFile);
        }

        // Export Cloth
        [UnityTest]
        public IEnumerator TestClothExport () {

            yield return TestScene(c_scene);
       }
```